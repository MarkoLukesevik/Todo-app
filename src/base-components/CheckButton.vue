<script setup lang="ts">
import { computed } from 'vue'

const props = defineProps<{
  isChecked: boolean
}>()

const emit = defineEmits<{
  (e: 'toggle', value: boolean): void
}>()

const checkButtonBackground = computed<string>(() => {
  return props.isChecked ? 'linear-gradient(hsl(192, 100%, 67%), hsl(280, 87%, 65%))' : ''
})
</script>

<template>
  <button
    class="check-button"
    :style="{ background: checkButtonBackground }"
    @click="emit('toggle', !isChecked)"
  >
    <img v-if="isChecked" src="/assets/images/icon-check.svg" alt="" />
  </button>
</template>

<style scoped lang="scss">
.check-button {
  display: flex;
  flex-direction: row;
  flex: 0 1 auto;
  align-items: center;
  justify-content: center;

  width: 20px;
  height: 20px;

  background-color: transparent;
  border: 1px solid #979797;
  border-radius: 50%;

  &:hover {
    background:
      linear-gradient(white, white) padding-box,
      linear-gradient(to right, hsl(192, 100%, 67%), hsl(280, 87%, 65%)) border-box;
    border: 1px solid transparent;
  }
}
</style>
