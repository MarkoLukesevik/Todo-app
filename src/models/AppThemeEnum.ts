export enum AppThemeEnum {
    Light = 0,
    Dark = 1
}
