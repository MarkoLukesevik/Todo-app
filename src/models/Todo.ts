export interface Todo {
  id: string
  title: string
  isComplete: false
}
