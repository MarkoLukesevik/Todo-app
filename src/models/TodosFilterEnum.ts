export enum TodosFilterEnum {
    All = 0,
    Active = 1,
    Completed = 2
}