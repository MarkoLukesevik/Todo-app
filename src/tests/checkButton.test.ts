import { describe, it, expect, beforeEach } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import { AppThemeEnum } from '@/models/AppThemeEnum';

import CheckButton from '@/base-components/CheckButton.vue';

describe('Test suite for check button component', () => {
    let wrapper: ReturnType<typeof shallowMount>;

    beforeEach(() => {
        wrapper = shallowMount(CheckButton, {
            props: {
                isChecked: false,
                appTheme: AppThemeEnum.Light,
            },
        });
    });

    it('Should match the snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('Should render correctly with default props', () => {
        const button = wrapper.find('button');
        expect(wrapper.find('img').exists()).toBe(false);
        
        expect(button.exists()).toBe(true);
        expect(button.classes()).toContain('check-button');
        expect(button.element.style.background).toBe('');
    });

    it('Should render image when isChecked prop is true', async () => {
        expect(wrapper.find('img').exists()).toBe(false);

        await wrapper.setProps({ isChecked: true });

        expect(wrapper.find('img').exists()).toBe(true);
    });

    it('Should add dark theme class when appTheme is Dark', async () => {
        await wrapper.setProps({ appTheme: AppThemeEnum.Dark });
        expect(wrapper.find('button').classes()).toContain('check-button-dark');
    });

    it('Should add no-border class when isChecked is true', async () => {
        await wrapper.setProps({ isChecked: true });
        expect(wrapper.find('button').classes()).toContain('check-button__no-border');
        expect(wrapper.find('img').exists()).toBe(true); // Image should be present
    });

    it('Should emit toggle event with the correct value when clicked', async () => {
        await wrapper.setProps({ isChecked: false });
        await wrapper.find('button').trigger('click');
        expect(wrapper.emitted('toggle')).toBeTruthy();
        expect(wrapper.emitted('toggle')?.[0]).toEqual([true]); // Click should toggle the value
    });
});