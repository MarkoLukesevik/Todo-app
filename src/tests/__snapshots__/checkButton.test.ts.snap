// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test suite for check button component > Should match the snapshot 1`] = `
<button
  class="check-button"
  data-v-26dbdb21=""
>
  <!--v-if-->
</button>
`;
